### DISCUSSION TITLE
404 error nuxt /api/auth/session using sidebase+prisma

### DISCUSSION THREAD

**Discussion by sTorba24** [vercel/community#4835] [Help]
### Summary

Hi! I've stack with the issue of using nuxt auth (sidebase) with prisma(not sure it is a point) on vercel

After i implemented it, i have an error 404 page visiting api/auth/session 

this part is implemented on server-side as it is in documentation

https://sidebase.io/nuxt-auth/getting-started/quick-start

Firstly i had an error because not set ORIGIN, but now i did it

https://sidebase.io/nuxt-auth/recipes/vercel-deployment

this auth route should be managed by served but seems it doesn't work in such way

I'm not so experienced using vercel so can't find a reason how to fix it.

On local it works fine without ORIGIN and SECRET env variables

Could you help me somehow>?

Getting the same error for all server parts
![photo_2023-11-13_10-58-57](https://github.com/vercel/community/assets/55200530/75bdf3a3-1227-4dbc-9eba-108ea3da06e1)

using /api/....

using /smth_server

it's not working for me, getting 404 page

Also i found and example of sidebase implementation, but oldest version and without prisma

and it's ok deployed
https://github.com/sidebase/nuxt-auth-example

env variables
![image](https://github.com/vercel/community/assets/55200530/b39842b9-250c-4eee-b5b1-bddace43ed7f)

i did it in the same way, with the same variables(it is from my project)
![image](https://github.com/vercel/community/assets/55200530/14510721-113c-4bfc-9685-69f6e23bd0bc)

here you can see my package.json file

{
  "name": "nuxt-prisma",
  "version": "2.0.0",
  "private": true,
  "scripts": {
    "dev": "npm run prisma:migrate:generate && nuxt dev --dotenv .env.dev",
    "build": "nuxt build",
    "start": "node .output/server/index.mjs",
    "generate": "nuxt generate",
    "prisma:migrate:dev": "npx prisma migrate dev",
    "prisma:migrate:generate": "npx prisma generate",
    "prisma:studio": "npx prisma studio"
  },
  "dependencies": {
    "@next-auth/prisma-adapter": "^1.0.7",
    "@prisma/client": "^5.5.2",
    "next-auth": "4.21.1",
    "nuxt": "3.8.1"
  },
  "devDependencies": {
    "@babel/eslint-parser": "7.23.3",
    "@nuxtjs/eslint-config": "12.0.0",
    "@nuxtjs/eslint-module": "4.1.0",
    "@sidebase/nuxt-auth": "^0.6.0-rc.1",
    "autoprefixer": "^10.4.16",
    "eslint": "8.53.0",
    "eslint-config-prettier": "9.0.0",
    "eslint-plugin-nuxt": "4.0.0",
    "eslint-plugin-prettier": "5.0.1",
    "postcss": "^8.4.31",
    "prettier": "3.0.3",
    "prisma": "5.5.2",
    "tailwindcss": "^3.3.5"
  },
  "prisma": {
    "seed": "node prisma/seed.js"
  }
}

here is my nuxt config part for auth

 auth: {
    isEnabled: true,
    baseURL: process.env.ORIGIN,
  },
  
  preventing messages about `origin` i found in doc that now it is baseURL, without origin+basePATH
  
  i'm using https://your-space-admin.vercel.app/api/auth or just https://your-space-admin.vercel.app/ as baseURL, no difference
  
  Locally everything is fine
  
  Sorry if i explained bad, it is my first experience

### Example

https://your-space-admin.vercel.app/

### Steps to Reproduce

just visit main page. it redirects to 404

**Comment by zoey-kaiser** [comment]
Hiya! 

From the NuxtAuth perspective. You are missing some key points in your `nuxt.config.ts`. Could you please go to https://sidebase.io/nuxt-auth/v0.6/configuration/nuxt-config and select your provider (I think in your case `authjs`, see the tabs in the code example) and fill out the remaining auth configuration values?

Thanks!

**Reply by sTorba24** [reply]
Thanks for the answer

changed auth config to 
auth: {
    // The module is enabled. Change this to disable the module
    isEnabled: true,
    provider: {
        type: 'authjs'
    },
    session: {
      enableRefreshPeriodically: true,
      enableRefreshOnWindowFocus: true
    },
    // globalAppMiddleware: {
    //   isEnabled: true
    // },
    baseURL: process.env.ORIGIN,
  },
  
Commented globalAppMiddleware, because it provokes failing deploy, also can't get why?
![image](https://github.com/vercel/community/assets/55200530/385e0597-2dbb-4388-9a88-f3aff73e9279)

even without it still not working :(

redirects from index page to 404
![image](https://github.com/vercel/community/assets/55200530/699ed212-5890-440f-9bfe-cb67a2320ae8)

**Reply by sTorba24** [reply]
you're right, seems problem isn't in auth, thanks
